### PR TITLE
marimo: init at 0.3.3

### PR DESCRIPTION
**Pull request by dmadisetti**
## Description of changes

Closing out my packaging request in #279873, by adding [`marimo`](https://docs.marimo.io/index.html) from [pypi packaging](https://pypi.org/project/marimo). Marimo is "A reactive notebook for Python — run reproducible experiments, execute as a script, deploy as an app, and version with git."

Github here: https://github.com/marimo-team/marimo

## Things done

 - Added myself as a maintainer (but happy to cede authority to anyone)
 - Added akshayka as maintainer- for reviewer reference they are the marimo lead
 - Added marimo package
Made sure it works!

image

I am also cross-posting this PR into their discord channel for upstream feedback.

---

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by akshayka** [comment]
@dmadisetti, thanks for creating this PR. I'm happy to be added as another maintainer. I am not familiar with Nix (though I have heard great things!) so it would be best if you stayed on as a co-maintainer if possible.

**Comment by TomaSajt** [comment]
Oh, I see now that the source given by `fetchPypi` includes all the frontend files, so no need to manually generate it with nodejs.

File: maintainers/maintainer-list.nix
Comment (TomaSajt) [inline maintainers/maintainer-list.nix:765]:
```suggestion
    github = "akshayk";
```

Comment (TomaSajt) [inline maintainers/maintainer-list.nix:765]:
```suggestion
  akshayka = {
    github = "akshayka";
```

I think that's the correct name

**Comment by TomaSajt** [comment]
By the way, is there any reason someone might want to use the package as a python package (so actually importing stuff from it) or is it just used as a framework for running the code?

If it's just for running, I'd say that you should just use `buildPythonApplication` directly, and move the package to `pkgs/by-name`

**Comment by dmadisetti** [comment]
Added akshayka as maintainer- for reviewer reference they are the marimo lead.

Yes! You can use marimo as a standalone package for building execution graphs.

Will fix maintainer typos and rebase- one second

**Comment by dmadisetti** [comment]
Bit of clarity, it's not just the app- you can use marimo for managine scripts or just import its UI tools in jupyter

File: pkgs/development/python-modules/marimo/default.nix
Comment (TomaSajt) [inline pkgs/development/python-modules/marimo/default.nix:49]:
```suggestion
    license = licenses.asl20;
    mainProgram = "marimo";
```
If an app has an obvious "main executable" then it should be named in `meta.mainProgram`

**Review by TomaSajt — CHANGES_REQUESTED** [review]
Looks fine for the most part, though there are some small things I think you should change.

**Comment by dmadisetti** [comment]
Oh woops, suggestions 4 days ago? Sorry I don't know why I didn't see them.

Good and reasonable suggestions, all implemented

File: pkgs/development/python-modules/marimo/default.nix
Comment (dmadisetti) [reply pkgs/development/python-modules/marimo/default.nix:46]: removed platforms.all

File: pkgs/development/python-modules/marimo/default.nix
Comment (TomaSajt) [inline pkgs/development/python-modules/marimo/default.nix:47]: I think a better description should be chosen for this.
I like the one in `README.md` the most:
"A reactive Python notebook that's reproducible, git-friendly, and deployable as scripts or apps"

Also, descriptions should not have a period at the end.

**Comment by TomaSajt** [comment]
> Oh woops, suggestions 4 days ago? Sorry I don't know why I didn't see them.

I think I just forgot to press the "Finish review" button until today.

**Review by TomaSajt — APPROVED** [review]
LGTM

I cannot merge this, as I do not have the permissions for that. If this PR gets lost among the others, we could send it into a "pull requests already reviewed" discussion.

**Review by ghost — COMMENTED** [review]
<details>
<summary>looks good. there are tests that should be run so can add `pytestCheckHook`. `setuptools` is part of the buildsystem so should be in `build-system` (alias for nativeBuildInputs). should also add an import check. ( i moved propogatedBuildInputs into `dependancies` they are equivalent.  click to see diff.
</summary>

```diff
diff --git a/pkgs/development/python-modules/marimo/default.nix b/pkgs/development/python-modules/marimo/default.nix
index b8a5d13474ff..d02f85896200 100644
--- a/pkgs/development/python-modules/marimo/default.nix
+++ b/pkgs/development/python-modules/marimo/default.nix
@@ -14,6 +14,7 @@
 , websockets
 , docutils
 , black
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
@@ -28,8 +29,11 @@ buildPythonPackage rec {
     hash = "sha256-l5QehU/LqEWb7ybKxace4sm6C9mcNlaHNOp55ExglpQ=";
   };
 
-  propagatedBuildInputs = [
+  build-system = [
     setuptools
+  ];
+
+  dependencies = [
     click
     jedi
     markdown
@@ -43,6 +47,14 @@ buildPythonPackage rec {
     black
   ];
 
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "marimo"
+  ];
+
   meta = with lib; {
     description = "A reactive Python notebook that's reproducible, git-friendly, and deployable as scripts or apps";
     homepage = "https://github.com/marimo-team/marimo";
```

**Comment by dmadisetti** [comment]
Thanks! Applied

**Review by ghost — APPROVED** [review]
LGTM. verified github ids match. will merge once tests complete (i verified aarch64-darwin)

**Pull request merged**
